dspace-api/src/main/java/org/dspace/curate/CurationTaskConsumer.java

```diff
@@ -0,0 +1,121 @@
+        }
+    }
+
+    /**
+     * Resolve and execute the configured curation task for this event's subject
+     * @param ctx   the current DSpace session
+     * @param event the content event
+     * @throws Exception
+     */
+    @Override
+    public void consume(Context ctx, Event event) throws Exception {
+        log.debug("Consumer Event " + event.toString());
+        ScriptConfiguration scriptToExecute = scriptService.getScriptConfiguration("curate");
+        DSpaceRunnable dSpaceRunnable = scriptService.createDSpaceRunnableForScriptConfiguration(scriptToExecute);
+        List<DSpaceCommandLineParameter> parameters = Arrays.asList(new DSpaceCommandLineParameter[] {
+            new DSpaceCommandLineParameter("-t", this.curationTaskName),
+            new DSpaceCommandLineParameter("-i", event.getSubjectID().toString())
+        });
+        List<String> args = new ArrayList<>(2 * parameters.size());
+        for (DSpaceCommandLineParameter parameter : parameters) {
+            args.add(parameter.getName());
+            if (parameter.getValue() != null) {
+                args.add(parameter.getValue());
+            }
+        }
+
+        if (scriptToExecute == null) {
+            log.error("Unable to find script curate");
+            throw new RuntimeException("The script for name: curate wasn't found");
+        }
+        try {
+            if (!scriptToExecute.isAllowedToExecute(ctx, parameters)) {
+                log.error("Current user is not eligible to execute script with name: curate and the specified " +
+                        "parameters " + StringUtils.join(parameters, ", "));
+                return;
+            }
+        } catch (IllegalArgumentException e) {
+            log.error("Illegal argument " + e.getMessage(), e);
+        }
+        log.debug("Initializing process " + scriptToExecute.getName() + " for user " + ctx.getCurrentUser().getID() +
+                "with parameters: " + StringUtils.join(parameters, " ") + ".");
+        ProcessDSpaceRunnableHandler processDSpaceRunnableHandler = new ProcessDSpaceRunnableHandler(
+                ctx.getCurrentUser(), scriptToExecute.getName(), parameters,
+                new HashSet<>(ctx.getSpecialGroups()));
+        try {
+            dSpaceRunnable.initialize(args.toArray(new String[0]), processDSpaceRunnableHandler, ctx.getCurrentUser());
+            processDSpaceRunnableHandler.schedule(dSpaceRunnable);
+        } catch (ParseException e) {
+            dSpaceRunnable.printHelp();
+            try {
+                processDSpaceRunnableHandler.handleException(
+                        "Failed to parse the arguments given to the script with name: " + scriptToExecute.getName()
+                                + " and args: " + StringUtils.join(parameters, " "), e
+                );
+            } catch (Exception re) {
+                // ignore re-thrown exception
+            }
+        }
```

Comment:
This whole section has been more or less duplicated from the `ScriptRestRespository`. I think it would be better to create proper utility functions here, and then handle exception properly and differently ( the runtime / exception should not be thrown inside a consumer, it could create some errors ).